### ISSUE TITLE
HAProxy spawning multiple processes when too many events occur in a row

### ISSUE THREAD

**Issue by mbentley** [ehazlett/interlock#32]
When performing a `docker rm -f <container>` and then immediately after doing a `docker run` on a container to replace it (like when doing an app deployment), interlock seems to be not able to keep up with the requests.  As a result, multiple haproxy instances end up running when the remove and create happen 10 times in a row.  In this case, I end up with 4 HAProxy pids running in the container:

```
UID                 PID                 PPID                C                   STIME               TTY                 TIME                CMD
root                15334               10629               1                   16:56               ?                   00:00:02            /usr/local/bin/interlock --swarm-url=tcp://54.152.22.32:3376 --swarm-tls-ca-cert=/etc/docker/ca.pem --swarm-tls-cert=/etc/docker/ser
ver.pem --swarm-tls-key=/etc/docker/server-key.pem --debug --plugin=haproxy start
root                15785               15334               0                   16:57               ?                   00:00:00            /usr/sbin/haproxy -D -f /proxy.conf -p /proxy.pid -sf 21
root                15793               15334               0                   16:57               ?                   00:00:00            /usr/sbin/haproxy -D -f /proxy.conf -p /proxy.pid -sf 25
root                15794               15334               0                   16:57               ?                   00:00:00            /usr/sbin/haproxy -D -f /proxy.conf -p /proxy.pid -sf 25
root                15797               15334               0                   16:57               ?                   00:00:00            /usr/sbin/haproxy -D -f /proxy.conf -p /proxy.pid -sf 36
```

Adding a delay, it ends up working as expected:

```
UID                 PID                 PPID                C                   STIME               TTY                 TIME                CMD
root                16476               10629               0                   17:02               ?                   00:00:00            /usr/local/bin/interlock --swarm-url=tcp://54.152.22.32:3376 --swarm-tls-ca-cert=/etc/docker/ca.pem --swarm-tls-cert=/etc/docker/ser
ver.pem --swarm-tls-key=/etc/docker/server-key.pem --debug --plugin=haproxy start
root                17442               16476               0                   17:06               ?                   00:00:00            /usr/sbin/haproxy -D -f /proxy.conf -p /proxy.pid -sf 58
```

This is when using the latest interlock:
`time="2015-04-11T17:09:27Z" level="info" msg="interlock running version=0.2.1 (cd41ba3)"`

**Comment by ehazlett** [comment]
Thanks!  This looks to be caused by the event stream firing faster than haproxy can restart causing multiple processes.  Digging.

**Comment by ehazlett** [comment]
This should be fixed in 6ecde26e6f
